wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java

```diff
@@ -123,6 +125,36 @@ protected Connection getValidatedConnection(ConnectionStore store, NamespacedId
     return connection;
   }
 
+  /**
+   * Utility method for executing an endpoint with common error handling built in.
+   * A response will always be sent after this method is called so the http responder should not be used after this.
+   * The endpoint logic should also not use the responder in any way.
+   *
+   * If the callable throws a {@link StatusCodeException}, the exception's status code and message will be used
+   * to create the response.
+   * If a {@link JsonSyntaxException} is thrown, a 400 response will be sent.
+   * If anything else if thrown, a 500 response will be sent.
+   * If nothing is thrown, the result of the callable will be sent as json.
+   *
+   * @param request the http request
+   * @param responder the http responder
+   * @param callable the endpoint logic to run
+   */
+  protected <T> void respond(HttpServiceRequest request, HttpServiceResponder responder, Callable<T> callable) {
```

Comment:
why do we need 2 methods one with namespace and one without?

Reply:
almost all endpoints are namespaced, but there are a couple that are not. They ones that are not use the method without a namespace, which avoids the unneeded lookup.